scripts/prep_data_for_model.R

```diff
@@ -130,18 +143,58 @@ gcs_upload(efficiency_summary,
            name = "jpe-model-data/efficiency_summary.csv",
            predefinedAcl = "bucketLevel")
 write_csv(efficiency_summary, "data/model-data/efficiency_summary.csv")
+# weekly release
+weekly_release <- standard_release |> 
+  select(stream, site, release_id, date_released, week_released, year_released, 
+         number_released, median_fork_length_released, flow_at_release, temperature_at_release, 
+         turbidity_at_release) |> 
+  mutate(week_released = ifelse(is.na(week_released), week(date_released), week_released),
+         year_released = ifelse(is.na(year_released), year(date_released), year_released)) |> 
+  group_by(stream, site, week_released, year_released) |> 
+  summarise(number_released = sum(number_released),
+            median_fork_length_released = median(median_fork_length_released, na.rm = T),
+            flow_at_release = mean(flow_at_release, na.rm = T),
+            temperature_at_release = mean(temperature_at_release, na.rm = T),
+            turbidity_at_release = mean(turbidity_at_release, na.rm = T)) |> 
+  mutate(across(everything(), ~replace(., is.nan(.), NA)))
+# weekly recapture
+weekly_recapture <- recapture_summary |> 
+  select(stream, site, release_id, date_released, week_released, year_released, 
+         number_recaptured, median_fork_length_recaptured) |> 
+  mutate(week_released = ifelse(is.na(week_released), week(date_released), week_released),
+         year_released = ifelse(is.na(year_released), year(date_released), year_released)) |> 
+  group_by(stream, site, week_released, year_released) |> 
+  summarise(number_recaptured = sum(number_recaptured),
+            median_fork_length_recaptured = median(median_fork_length_recaptured, na.rm = T))
+# weekly efficiency
+# this weekly summary assumes fish released in week 1 are caught in week 1
+weekly_efficiency <- left_join(weekly_release, weekly_recapture)
```

Comment:
Did we check here for weeks where fish were released but not recaptured? Worth checking and then adding 0 recaps to the weekly tables if they exist.

Reply:
Doesn't look like there are any NA weekly recaps so we are good. I did notice there are 24 more rows in the recaptured table than there are in the release table. Since you are left joining on the release table we aren't capturing these recaptures that do not come with a release week. I will look a little more into what these weeks look like...